bindings_node/src/mls_client.rs

```diff
@@ -114,56 +118,75 @@ impl NapiClient {
       signature_bytes.deref().to_vec(),
     ));
 
-    signature_request
-      .add_signature(signature)
-      .await
-      .map_err(|e| Error::from_reason(format!("{}", e)))?;
-
     self
-      .inner_client
-      .register_identity(signature_request)
-      .await
-      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+      .signatures
+      .insert(SignatureType::RecoverableEcdsaSignature, signature);
 
-    return Ok(());
+    Ok(())
   }
 
   #[napi]
-  pub async fn register_erc1271_identity(
-    &self,
+  pub fn add_erc1271_signature(
+    &mut self,
     signature_bytes: Uint8Array,
+    chain_id: String,
+    account_address: String,
     chain_rpc_url: String,
+    block_number: BigInt,
   ) -> Result<()> {
     if self.is_registered() {
       return Err(Error::from_reason(
         "An identity is already registered with this client",
       ));
     }
 
-    let mut signature_request = match self.inner_client.identity().signature_request() {
-      Some(signature_req) => signature_req,
-      None => return Err(Error::from_reason("No signature request found")),
-    };
     let signature_text = match self.signature_text() {
       Some(text) => text,
       None => return Err(Error::from_reason("No signature text found")),
     };
 
-    let signature = Box::new(
-      Erc1271Signature::new_with_rpc(
-        signature_text,
-        signature_bytes.deref().to_vec(),
-        self.account_address.clone(),
-        chain_rpc_url,
-      )
-      .await
-      .map_err(|e| Error::from_reason(format!("{}", e)))?,
-    );
+    let account_id = AccountId::new(chain_id, account_address);
 
-    signature_request
-      .add_signature(signature)
-      .await
-      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+    let signature = Box::new(Erc1271Signature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+      account_id,
+      chain_rpc_url,
+      block_number.get_u64().1,
```

Comment:
If `BigInt` isn't doing much, would it make sense to just use `u64`?

Reply:
`BigInt` is required for larger numbers.

see: https://napi.rs/docs/concepts/values#bigint